content/en/blog/general/generate-vaps/index.md

```diff
@@ -0,0 +1,357 @@
+spec:
+  failurePolicy: Fail
+  matchConstraints:
+    matchPolicy: Equivalent
+    namespaceSelector: {}
+    objectSelector: {}
+    resourceRules:
+    - apiGroups:
+      - apps
+      apiVersions:
+      - v1
+      operations:
+      - CREATE
+      - UPDATE
+      resources:
+      - deployments
+      scope: '*'
+  validations:
+  - expression: '!has(object.spec.template.spec.volumes) || object.spec.template.spec.volumes.all(volume,
+      !has(volume.hostPath))'
+    message: HostPath volumes are forbidden. The field spec.template.spec.volumes[*].hostPath
+      must be unset.
+  variables: null
+status:
+  observedGeneration: 1
+  typeChecking: {}
+```
+
+```bash
+$ kubectl get validatingadmissionpolicybindings disallow-host-path-binding -o yaml
+apiVersion: admissionregistration.k8s.io/v1beta1
+kind: ValidatingAdmissionPolicyBinding
+metadata:
+  creationTimestamp: "2023-09-12T11:42:13Z"
+  generation: 1
+  labels:
+    app.kubernetes.io/managed-by: kyverno
+  name: disallow-host-path-binding
+  ownerReferences:
+  - apiVersion: kyverno.io/v1
+    kind: ClusterPolicy
+    name: disallow-host-path
+    uid: e540d96b-c683-4380-a84f-13411384241a
+  resourceVersion: "11292"
+  uid: 2fec35c3-8a8c-42a7-8a02-a75e8882a01e
+spec:
+  policyName: disallow-host-path
+  validationActions:
+  - Deny
+```
+
+Now, let’s try deploying an app that uses a hostPath:
+
+```bash
+kubectl apply -f - <<EOF
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx
+spec:
+  replicas: 2
+  selector:
+    matchLabels:
+      app: nginx
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      containers:
+      - name: nginx-server
+        image: nginx
+        volumeMounts:
+          - name: udev
+            mountPath: /data
+      volumes:
+      - name: udev
+        hostPath:
+          path: /etc/udev
+EOF
+```
+
+As expected, the deployment creation is rejected by the API server and not by the Kyverno admission controller.
+
+```bash
+The deployments "nginx" is invalid:  ValidatingAdmissionPolicy 'disallow-host-path' with binding 'disallow-host-path-binding' denied request: HostPath volumes are forbidden. The field spec.template.spec.volumes[*].hostPath must be unset.
+```
+
+If either the ValidatingAdmissionPolicy or the binding is deleted/updated for some reason, the controller is responsible for reverting it.
+
+Let’s try deleting the ValidatingAdmissionPolicy.
+
+```bash
+$ kubectl delete validatingadmissionpolicy disallow-host-path 
+validatingadmissionpolicy.admissionregistration.k8s.io "disallow-host-path" deleted
+
+$ kubectl get validatingadmissionpolicy 
+NAME                                  VALIDATIONS   PARAMKIND   AGE
+disallow-host-path    1                        <unset>        11s
+```
+
+In addition, you can update the Kyverno policy, and the controller will re-generate the ValidatingAdmissionPolicy accordingly. For example, you can change the Kyverno policy to match statefulsets too.
```

Comment:
You could show a `kubectl patch` command to simply add this to the existing policy.